SmartDeviceLinkTests/UtilitiesSpecs/Touches/SDLTouchManagerSpec.m

```diff
@@ -85,15 +87,26 @@ + (void)testTouchesWithTimeout:(CGFloat)timeoutTime
 QuickSpecBegin(SDLTouchManagerSpec)
 
 describe(@"SDLTouchManager Tests", ^{
-    __block SDLTouchManager *touchManager;
-
+    __block SDLTouchManager *touchManager = nil;
+    __block void (^unloadTouchManager)(void) = ^() {
+        if (touchManager) {
+            //FIXIT: SDLTouchManager must unsubscribe from notifications
+            [[NSNotificationCenter defaultCenter] removeObserver:touchManager];
+            touchManager = nil;
+        }
+    };
+    
     context(@"initializing", ^{
         it(@"should correctly have default properties", ^{
+            expect(touchManager).to(beNil());
             SDLTouchManager* touchManager = [[SDLTouchManager alloc] initWithHitTester:nil];
             expect(touchManager.touchEventDelegate).to(beNil());
             expect(@(touchManager.tapDistanceThreshold)).to(equal(@50));
             expect(@(touchManager.tapTimeThreshold)).to(beCloseTo(@0.4).within(0.0001));
             expect(@(touchManager.isTouchEnabled)).to(beTruthy());
+            //FIXIT: SDLTouchManager must unsubscribe from notifications
+            [[NSNotificationCenter defaultCenter] removeObserver:touchManager];
+            touchManager = nil;
```

Comment:
This brakes tests. I'd better keep it as it is.
BTW: I tried to add to SDLTouchManager.m the following method though it did not help:
-(void)dealloc { [[NSNotificationCenter defaultCenter] removeObserver:self]; }

Besides SDL supports iOS-8 and up. In iOS-8 I must unsubscribe from notifications otherwise it crashes when calling deallocated object. I must unsubscribe from notifications somewhere, anywhere.

Reply:
replaced by unloadTouchManager();
I must unsubscribe from notifications